tests/test_types.py

```
@@ -356,6 +356,53 @@ def test_invalid_disk_mode_rejected(self, tmp_path: Path) -> None:
                 disk_mode="snapshot",
             )
 
+    def test_windows_guest_requires_firmware_boot(self, tmp_path: Path) -> None:
+        """Windows guests must boot via OVMF firmware — no direct-kernel path."""
+        kernel = tmp_path / "vmlinux"
+        rootfs = tmp_path / "win11.qcow2"
+        kernel.touch()
+        rootfs.touch()
+
+        with pytest.raises(ValidationError, match="windows.*requires boot_mode='firmware'"):
```

Ruff (0.15.14)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)